### PR TITLE
feat: Allows to wrap distributor Tee

### PR DESCRIPTION
**Pull request by cyriltovena**
**What this PR does / why we need it**:

This is going to be useful when we we want to tee to multiple destination and avoid the original Tee being replaced.

cc @MasslessParticle 

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/distributor/tee_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package distributor
+
+import (
+	"testing"
+
+	"github.com/grafana/loki/pkg/push"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockedTee struct {
```
Comment (na--) [inline pkg/distributor/tee_test.go:10]: Nit, but there is already a `mockTee` here: https://github.com/grafana/loki/blob/111c19b38147837041c6cc7c7483abe00c0c1599/pkg/distributor/distributor_test.go#L1268

Comment (cyriltovena) [reply pkg/distributor/tee_test.go:10]: yeah it's a bit different.

**Review by MasslessParticle — APPROVED** [review]
This makes sense

**Comment by cyriltovena** [comment]
> This makes sense

Thank you !

**Pull request merged**